glide-core/src/client/value_conversion.rs

```diff
@@ -379,9 +368,89 @@ pub(crate) fn convert_to_expected_type(
+        /* RESP2 response
+        1) 1) "library_name"
+           2) "mylib1"
+           3) "engine"
+           4) "LUA"
+           5) "functions"
+           6) 1) 1) "name"
+                 2) "myfunc1"
+                 3) "description"
+                 4) (nil)
+                 5) "flags"
+                 6) (empty array)
+              2) 1) "name"
+                 ...
+        2) 1) "library_name"
+           ...
+
+        RESP3 response
+        1) 1# "library_name" => "mylib1"
+           2# "engine" => "LUA"
+           3# "functions" =>
+              1) 1# "name" => "myfunc1"
+                 2# "description" => (nil)
+                 3# "flags" => (empty set)
+              2) 1# "name" => "myfunc2"
+                 ...
+        2) 1# "library_name" => "mylib2"
+           ...
+        */
+        ExpectedReturnType::ArrayOfMaps(type_of_map_values) => match value {
+            // empty array, or it is already contains a map (RESP3 response) - no conversion needed
+            Value::Array(ref array) if array.is_empty() || matches!(array[0], Value::Map(_)) => {
+                Ok(value)
+            }
+            Value::Array(array) => convert_array_of_flat_maps(array, Some(*type_of_map_values)),
+            // cluster (multi-node) response - go recursive
+            Value::Map(map) => convert_map_entries(
+                map,
+                Some(ExpectedReturnType::BulkString),
+                Some(ExpectedReturnType::ArrayOfMaps(type_of_map_values)),
+            ),
+            // Due to recursion, this will convert every map value, including simple strings, which we do nothing with
+            Value::BulkString(_) | Value::SimpleString(_) | Value::VerbatimString { .. } => {
+                Ok(value)
+            }
+            _ => Err((ErrorKind::TypeError, "Response couldn't be converted").into()),
+        },
+        // Not used for a command, but used as a helper for `FUNCTION LIST` to process the inner map.
+        // It may contain a string (name, description) or set (flags), or nil (description).
+        // The set is stored as array in RESP2. See example for `ArrayOfMaps` above.
+        ExpectedReturnType::StringOrSet => match value {
+            Value::Nil => Ok(value),
+            Value::Array(_) => convert_to_expected_type(value, Some(ExpectedReturnType::Set)),
+            Value::BulkString(_) | Value::SimpleString(_) | Value::VerbatimString { .. } => {
+                Ok(value)
+            }
+            _ => Err((ErrorKind::TypeError, "Response couldn't be converted").into()),
+        },
     }
 }
 
+/// Similar to [`convert_array_to_map`], but converts keys and values to the given types inside the map.
+/// The input data is [`Value::Map`] payload, the output is the new [`Value::Map`].
+fn convert_map_entries(
+    map: Vec<(Value, Value)>,
+    key_expected_return_type: Option<ExpectedReturnType>,
```

Comment:
nit: rename it as in Andrew's PR to key_type and value_type
https://github.com/aws/glide-for-redis/pull/1494/files#diff-d19cefadf2507886101854bc083f6dd9ab5c6a49549e2c2e998a209b4ded03a0R427